chain-signatures/node/src/mesh/connection.rs

```diff
@@ -39,11 +40,30 @@ impl Pool {
         let mut status = self.status.write().await;
         let mut participants = Participants::default();
         for (participant, info) in connections.iter() {
-            let Ok(resp) = self.http.get(format!("{}/state", info.url)).send().await else {
+            let Ok(Ok(url)) = Url::parse(&info.url).map(|url| url.join("/state")) else {
```

Comment:
We should make `ParticipantInfo::url` of type `Url` so we don't have to do this parsing on the fly. Doesn't have to be done in this PR

Reply:
Sounds good

Reply:
Will address in a separate PR